### ISSUE TITLE
[Improvement]: Transform Asynchronous Tasks to MessageHandler

### ISSUE THREAD

**Issue by solverat** [pimcore/pimcore#9810]
### Improvement description

I would like to suggest an improvement regarding asynchronous tasks:

**tl;dr:** Every asynchronous task in PIMCORE should be registered as a message handler. 

***

- There are still several classes which are using the old-fashioned way to execute process in background by using the deprecated method `runPhpScriptInBackground`: https://github.com/pimcore/pimcore/search?q=runPhpScriptInBackground&type=code
- All Maintenance Tasks
- Thumbnail Generation Tasks (And new Features like #9758)

## Default Configuration
Using the doctrine transport as an asynchronous transport should be the default, but allows user to change it to any supported transport like AMQP:

```yaml
framework:
    messenger:
        transports:
            async: 'doctrine://default'
```

## Process
Symfony recommends a supervisor configuration, but it should be possible to still use the cron process

```bash
* * * * * bin/console messenger:consume async --memory-limit=128M
``` 

## Benefits
- Using 100% Symfony standards
- Simplify Implementation of custom (long running) processes 
- Allow to switch transport without changing business logic
- Improved Error-Management

**Comment by brusch** [comment]
Duplicate, see #6371
@solverat would you like to contribute on that?

**Comment by dpfaffenbauer** [comment]
@brusch I am happy to help here if you still need help

**Comment by brusch** [comment]
@dpfaffenbauer yes, that would be very nice. Any particular topic you have in mind to start with?

**Comment by dpfaffenbauer** [comment]
@brusch I would start with moving stuff from runInBackground to a Messenger Queue. The queue would be just a doctrine handler

**Comment by brusch** [comment]
@dpfaffenbauer sounds good! Yes, our plan was also to use the doctrine handler.
